code/feature_extraction/FeatureExtractor.py

```diff
@@ -55,91 +54,210 @@ def __get_targets(self, truth_classes: pd.Series) -> pd.Series:
         nc_article_paragraphs = Util.count_chars(article_paragraphs)
 
         # Calculate num words
-        nw_post_title = len(w_post_title)
-        nw_post_image = self.wordtools.count_words(image_text)
-        nw_article_keywords = self.wordtools.count_words(article_keywords)
-        nw_article_description = self.wordtools.count_words(article_description)
-        nw_article_title = self.wordtools.count_words(article_title)
-        # Unused: nw_article_paragraphs = self.wordtools.count_words(article_paragraphs)
+        nw_post_title    = len(w_post_title)
+        nw_post_image    = len(w_post_image)
+        nw_article_kw    = len(w_article_kw)
+        nw_article_descr = len(w_article_descr)
+        nw_article_title = len(w_article_title)
+        # Note: num words article paragraphs currently not supported!
 
         # Calculate num formal words
-        nwf_post_title = len(wf_post_title)
+        nfw_post_title    = len(fw_post_title)
+        nfw_post_image    = len(fw_post_image)
+        nfw_article_kw    = len(fw_article_kw)
+        nfw_article_descr = len(fw_article_descr)
+        nfw_article_title = len(fw_article_title)
+
+        # Calculate num stop words
+        nsw_post_title    = len(sw_post_title)
+        nsw_post_image    = len(sw_post_image)
+        nsw_article_kw    = len(sw_article_kw)
+        nsw_article_descr = len(sw_article_descr)
+        nsw_article_title = len(sw_article_title)
+
+        # Calculate PoS features
+        pos_nn_post_title = Util.count_tags(pos_post_title, {'NN'})
 
         # ------
+        # TODO: remove amount of "duplicate" code here
+
+        # Number of characters
+        features['numChars_post_title']         = nc_post_title
+        features['numChars_post_image']         = nc_post_image
+        features['numChars_article_keywords']   = nc_article_kw
+        features['numChars_article_desc']       = nc_article_descr
+        features['numChars_article_title']      = nc_article_title
+        features['numChars_article_paragraphs'] = nc_article_paragraphs
+
+        # Number of words
+        features['numWords_post_title']       = nw_post_title
+        features['numWords_post_image']       = nw_post_image
+        features['numWords_article_keywords'] = nw_article_kw
+        features['numWords_article_desc']     = nw_article_descr
+        features['numWords_article_title']    = nw_article_title
+        # features['numWords_article_paragraphs'] = nw_article_paragraphs
+
+        # Number of formal words
+        features['numFormalWords_post_title']    = nfw_post_title
+        features['numFormalWords_post_image']    = nfw_post_image
+        features['numFormalWords_article_kw']    = nfw_article_kw
+        features['numFormalWords_article_descr'] = nfw_article_descr
+        features['numFormalWords_article_title'] = nfw_article_title
+        # features['numFormalWords_article_paragraphs'] = nfw_article_paragraphs
```

Comment:
why are these commented out? Is that because paragraphs contain multiple sentences (thus requiring an average over the sentences)?

Reply:
Yes! I'll add an issue to fix this next week (if we want to)